src/libraries/System.Private.CoreLib/src/System/Number.NumberToFloatingPointBits.cs

```diff
@@ -1006,11 +1004,19 @@ private static uint DigitsToUInt32(byte* p, int count)
             Debug.Assert((1 <= count) && (count <= 9));
 
             byte* end = (p + count);
-            uint res = (uint)(p[0] - '0');
+            uint res = 0;
 
-            for (p++; p < end; p++)
+            // parse batches of 8 digits with SWAR
+            while (end - p >= 8)
```

Comment:
```suggestion
            while (p <= end - 8)
```
This way `end - 8` is constant for the loop, thus avoiding repeated calculation of `end - p`.